src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -83,23 +81,44 @@ private AsyncDocumentClient getAsyncDocumentClient() {
     public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
     }
 
+    private void validate(String collectionName) {
+        Assert.hasText(collectionName, "collection should not be null, empty or only whitespaces.");
+    }
+
+    private void validate(DocumentQuery query, String collectionName, Class<?> entityClass,
+                          List<String> partitionKeyNames) {
+        Assert.notNull(query, "DocumentQuery should not be null.");
+        Assert.hasText(collectionName, "collection should not be null, empty or only whitespaces.");
+        Assert.notNull(entityClass, "domainClass should not be null.");
+        Assert.notNull(partitionKeyNames, "partitionKeyNames should not be null.");
```

Comment:
Can we reuse the third  validate here?

Reply:
yeah, make sense. I will apply the validate in another PR and reuse the implementation.